zipline/pipeline/factors/statistical.py

```diff
@@ -27,13 +28,26 @@ def __new__(cls,
                 target_slice,
```

Comment:
That would be good for the regression factor. For the correlation one, I'm thinking `base_factor` and `target`